projects/components/src/action-menu/action-menu.component.ts

```diff
@@ -65,14 +66,16 @@ export class ActionMenuComponent<R, T> {
      */
     shouldDisplayContextualActionsDropdown = false;
 
-    private _actionDisplayConfig: ActionDisplayConfig = DEFAULT_ACTION_DISPLAY_CONFIG;
+    private _actionDisplayConfig: ActionDisplayConfig = getCopyOfActionDisplayConfig(DEFAULT_ACTION_DISPLAY_CONFIG);
     /**
      * Display configuration of both static and contextual actions
      * If null or undefined is passed, default config {@link _actionDisplayConfig} is used
      */
     @Input() set actionDisplayConfig(config: ActionDisplayConfig) {
-        Object.keys(config || {}).forEach(
-            key => (this._actionDisplayConfig[key] = config[key] ? config[key] : DEFAULT_ACTION_DISPLAY_CONFIG[key])
+        const copyOfConfig = getCopyOfActionDisplayConfig(config || {});
+        const copyOfDefaultConfig = getCopyOfActionDisplayConfig(DEFAULT_ACTION_DISPLAY_CONFIG);
```

Comment:
Instead of duplicating code to get a copy, create a function to get a new instance every time

```
// Hard code objects returned will always be new instances
getDefaultActionDisplayConfig() {

}
```